doc/nrf/ug_bootloader.rst

```diff
@@ -7,212 +7,192 @@ Secure bootloader chain
+The :ref:`bootloader capabilities table <app_bootloaders_support_table>` indicates the bootloaders that are valid for use as an upgradable bootloader.
 
-     west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- -Db0_OVERLAY_CONFIG=overlay-minimal-size.conf -DCONFIG_SECURE_BOOT=y
+.. _upgradable_bootloader_presigned_variants:
 
+Pre-signed variants
+-------------------
 
-Adding MCUboot as an upgradable bootloader
-==========================================
+When programming an upgradable bootloader, the build system is able to automatically generate pre-signed variants of the image for both slots when compiling with the :option:`CONFIG_BUILD_S1_VARIANT` option.
+An OTA update package of the upgradable bootloader contains both images, since it might be unknown which slot is available for upgrade.
+See the :ref:`bootloader_flash_layout` of the |NSIB| for more details.
 
-To add MCUboot as upgradable bootloader to your application, set :option:`CONFIG_BOOTLOADER_MCUBOOT`.
-|how_to_configure|
+Therefore, these variants are necessary to perform the upgrade procedure of the bootloader through the :ref:`lib_fota_download` library.
 
-To make MCUboot upgradable, you must also add the immutable bootloader.
-Set option :option:`CONFIG_SECURE_BOOT` to do this.
+Additionally, building and programming these variants acts as a backup mechanism in case the image in the primary slot becomes corrupted, for example from a bit-flip, as the immutable bootloader can invalidate the corrupt image and boot into a valid one.
 
 .. note::
-   It is possible to include this bootloader without the immutable bootloader.
-   In this case, MCUboot will act as an immutable bootloader.
 
-It is possible for MCUboot to use the cryptographic functionality exposed by the immutable bootloader, reducing the flash usage for MCUboot to less than 16 kB.
-To enable this configuration, apply the :file:`overlay-minimal-external-crypto.conf` Kconfig overlay file for the MCUboot image.
-This can be done in the following way:
-
-* Using cmake::
-
-     cmake -GNinja -DBOARD=nrf52840dk_nrf5840 -Dmcuboot_OVERLAY_CONFIG=overlay-minimal-external-crypto.conf -DCONFIG_SECURE_BOOT=y -DCONFIG_BOOTLOADER_MCUBOOT=y ../
-
-* Using west::
-
-     west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- -Dmcuboot_OVERLAY_CONFIG=overlay-minimal-external-crypto.conf -DCONFIG_SECURE_BOOT=y -DCONFIG_BOOTLOADER_MCUBOOT=y
-
-See :doc:`mcuboot:index` for information about the default implementation of MCUboot.
-:ref:`mcuboot:mcuboot_ncs` gives details on the integration of MCUboot in |NCS|.
-
-You can configure MCUboot by setting configuration options for the ``mcuboot`` child image.
+   Building a pre-signed variant for slot 1 will not change the flash memory partition layout for an application, as the slot is already taken into account by default.
+   Instead, the flash memory content of slot 1 in the built hex file will be populated by an image instead of being erased or left as-is.
```

Comment:
I don't 100% understand the meaning of this. Who doesn't change the partition layout, and who populates the slot?

Reply:
Agreed, the wording needed to be updated. I changed up this whole section about pre-signed variants and moved some relevant information into the NSIB README. Please have a look at the latest version and see if it is better about this.